docs/configuration.md

```diff
@@ -1 +1,392 @@
+
+## Custom Resource Spec
+
+The desired state of resources created by the Backstage Operator is defined in the Backstage Custom Resource Spec. Here’s an example of a simple Backstage CR:
+
+  ```yaml
+apiVersion: rhdh.redhat.com/v1alpha2
+kind: Backstage
+metadata:
+  name: mybackstage
+spec:
+  application:
+    appConfig:
+      configMaps:
+        - name: my-app-config
+    extraEnvs:
+      secrets:
+        - name: my-secrets
+  ```
+
+This Custom Resource defines a Backstage instance called **mybackstage** and also:
+  - Adds additional app-config stored in the **my-app-config** ConfigMap.
+  - Adds some extra environment variables stored (as key-value pairs) in the Secret called **my-secrets**.
+
+For API version **v1alpha2** (Operator version **0.3.x**), the Backstage CR Spec contains the following top-level elements:
+
+  * [application](#application-configuration)
+  * [deployment](#deployment-configuration)
+  * [database](#local-database-configuration)
+  * [rawRuntimeConfig](#raw-configuration)
+  
+  ### Application Configuration
+  
+  This section explains how the Backstage Application is configured inside the container.
+  
+  #### app-config
+  
+  As documented in the [Backstage documentation](https://backstage.io/docs/conf/writing/), the Backstage application is configured with one or more app-config files, which are merged from the first to the last. The Operator can contribute to the app-config list by mounting the ConfigMap defined in `default-config/app-config.yaml`, as specified in the [Default Configuration](#default-configuration) section. Additionally, it is possible to define an array of external, user-created ConfigMaps located in the same namespace as the Backstage CR using `spec.application.appConfig`.
+
+For example, consider the following ConfigMaps containing Backstage app-config configuration:
+
+  ```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-app-config
+data:
+  my-app-config.yaml: |
+    # Some fragment of app-config here
+  ```
+
+This app-config file can be mounted to the Backstage container as follows:
+
+  ```yaml
+spec:
+  application:
+    appConfig:
+      mountPath: /my/path
+      configMaps:
+        - name: my-app-config
+  ```
+
+In this example, additional app-config is defined in the ConfigMaps called **my-app-config**, and it is mounted as YAML files to the `/my/path` directory in the Backstage container:
+
+  ```
+  /my/path/my-app-config.yaml
+  ```
+
+The ConfigMap key/value defines the file name and content, and these app-configs will be applied in the same order as they are declared in the CR. Therefore, as per this example, two ConfigMaps should be created in the namespace as a prerequisite. Then the Operator will create the files `/my/path/app-config1.yaml` and `/my/path/app-config2.yaml`, adding them to the end of the Backstage command-line arguments as follows:
```

Comment:
> Therefore, as per this example, two ConfigMaps should be created in the namespace as a prerequisite. Then the Operator will create the files `/my/path/app-config1.yaml` and `/my/path/app-config2.yaml`, adding them to the end of the Backstage command-line arguments as follows:

These sentences no longer seem consistent with the example provided. I guess they can be removed.

Reply:
fixed